src/stock.rs

```diff
@@ -200,40 +199,30 @@ impl Context {
         from_id: u32,
     ) -> String {
         let insert1 = if id == StockMessage::MsgAddMember || id == StockMessage::MsgDelMember {
-            unsafe {
-                let param1_c = CString::new(param1.as_ref()).unwrap();
-                let contact_id = dc_lookup_contact_id_by_addr(self, param1_c.as_ptr());
-                if contact_id != 0 {
-                    let contact = dc_get_contact(self, contact_id);
-                    let displayname = dc_contact_get_name_n_addr(contact);
-                    let ret = to_string(displayname);
-                    free(contact as *mut libc::c_void);
-                    free(displayname as *mut libc::c_void);
-                    ret
-                } else {
-                    param1.as_ref().to_string()
-                }
+            let contact_id = Contact::lookup_id_by_addr(self, param1.as_ref());
+            if contact_id != 0 {
+                Contact::get_by_id(self, contact_id)
+                    .map(|contact| contact.get_name_n_addr())
+                    .unwrap_or_default()
+            } else {
+                param1.as_ref().to_string()
             }
         } else {
             param1.as_ref().to_string()
         };
+
         let action = self.stock_string_repl_str2(id, insert1, param2.as_ref().to_string());
         let action1 = action.trim_end_matches('.');
         match from_id {
             0 => action,
             1 => self.stock_string_repl_str(StockMessage::MsgActionByMe, action1), // DC_CONTACT_ID_SELF
-            _ => unsafe {
-                let contact = dc_get_contact(self, from_id);
-                let displayname = dc_contact_get_display_name(contact);
-                let ret = self.stock_string_repl_str2(
-                    StockMessage::MsgActionByUser,
-                    action1,
-                    as_str(displayname),
-                );
-                free(contact as *mut libc::c_void);
-                free(displayname as *mut libc::c_void);
-                ret
-            },
+            _ => {
+                let displayname = Contact::get_by_id(self, from_id)
+                    .map(|contact| contact.get_name_n_addr())
```

Comment:
The logic used to be `contact.get_display_name()` here instead of `.get_name_n_addr()`.  Why did it change?

Reply:
probably because I can't read